documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
-General
--------
+# General
 
-### Partial Message - Request/Reply
+## KeepAlive - Request/Reply
 
-1.  Begin/Middle/End - Enum (1 byte)
-2.  Data
-A partial message is one that was broken up when being sent over the wire. (For any reason). The whole message is reconstructed by reading the partial messages in sequence and assembling them into a whole. It is not valid to attempt to start a new partial message before completing the previous one.
+# Reading
 
-### KeepAlive - Request/Reply
+## Read Segment - Request
 
-1.  Data - uninterpreted data of the length of the message. (Usually 0 bytes)
+1. Segment to read: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+2. Offset to read from: Long (8 bytes).
+3. Suggested Length of Reply: Integer (4 bytes). The clients can request for the required length to server (but the server may allot a different number of bytes).
+4. `delegationToken`: This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-Reading
--------
 
-### Read Segment - Request
+## Segment Read - Reply
 
-1.  Segment to read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset to read from - Long (8 bytes)
-3.  Suggested Length of Reply - int (4 bytes)
-    1.  This is how much data the client wants. They won't necessarily get that much.
+1.  Segment that was read: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+2.  Offset that was read from: Long (8 bytes).
+3.  Is at Tail: Boolean (1 bit).
+4.  Is at `EndOfSegment`: 1 bit.
+5.  `Data`: Binary (remaining length in the message).
 
-### Segment Read - Reply
+The client requests to read from a particular Segment at a particular Offset. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server decides on more or less data to give the client than it asked for, in as many replies as it sees fit.
 
-1.  Segment that was read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset that was read from - Long (8 bytes)
-3.  Is at Tail - Boolean (1 bit)
-4.  Is at EndOfSegment - (1 bit)
-5.  Data - Binary (remaining length in message)
+# Appending
 
-The client requests to read from a particular stream at a particular offset, it then receives one or more replies in the form of SegmentRead messages. These contain the data they requested (assuming it exists). The server may decided to give the client more or less data than it asked for, in as many replies as it sees fit.
+## Setup Append - Request
+1.  `RequestId`: Long (8 bytes). (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  `writerId`: UUID (16 bytes) identifies the requesting appender.
+3.  Segment to append: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
```

Comment:
Please fix formatting, here and anywhere else it is inconsistent.

Reply:
Accepted suggestion